### ISSUE TITLE
Is the mod supposed to work with a shaderpack ? [1.12.2]

### ISSUE THREAD

**Issue by khalen56** [tr7zw/EntityCulling#141]
So far I'm really happy with this mod, it greatly improves performances on a world with a lot of signpictures.

I just have a question, is the mod supposed to cull entities with shaders enabled ?
Without shaders, the debug shows me around 3K entities skipped, and last pass is always around 20ms.
With shaders, 0 entity skipped and last pass is always around 1 or 2ms.

Tested with OptiFine_1.12.2_HD_U_G5 and Chokapic13_V9 shaderpacks

**Comment by tr7zw** [comment]
Yes it does work with shaderpacks. Noteworthy that I don't test with Optifine since Sodium replaces that. On 1.12.2 culling is also turned off in f5 mode, as the game doesn't even know where the camera is located.

**Comment by khalen56** [comment]
![image](https://github.com/tr7zw/EntityCulling/assets/6369842/ea2e5a25-8464-432e-84b5-72a5b9a800ed)
![image](https://github.com/tr7zw/EntityCulling/assets/6369842/129f9df8-f03a-40e8-b71c-3ad9621895cf)

Any clue why it seems to render all entites with a shader on my setup ?

**Comment by tr7zw** [comment]
Possible that 1.12.2 optifine is weird. As I said, I don't test optifine, but last time I checked it worked fine with optifine shaders in 1.16+. Possible that 1.12.2 optifine renders shaders differently than in "newer" versions. In that case optifine 1.12.2 shaders would be unsupported.

**Comment by khalen56** [comment]
I've been checking the code, it seems that in CullTask.java, the noCulling boolean seems to change between true and false even when not in 3rd person when a shader is active.
Forcing the boolean to false make the culling working when a shader is active, that's weird 🤔

**Comment by tr7zw** [comment]
Don't have the source open rn, but the noCulling flag on each entity is a vanilla thing. Possible that optifine forces that to true for shadows.

**Comment by khalen56** [comment]
I'm talking of this line https://github.com/tr7zw/EntityCulling/blob/1.12/EntityCulling-Forge/src/main/java/dev/tr7zw/entityculling/CullTask.java#L58

It's just checking for the third person camera, so shaders in 1.12.2 seems to mess with that.

Also, it is interesting to see that there are two times more skipped entities with a shader. Maybe that can explains poor performances on 1.12.2 with shaders
![2024-03-15_20 46 57](https://github.com/tr7zw/EntityCulling/assets/6369842/8b5a8a04-1c8b-46cf-af14-d0a2f3ffa66d)
![2024-03-15_20 49 15](https://github.com/tr7zw/EntityCulling/assets/6369842/7a366b11-d384-4062-bf84-d5a711ecc85a)

**Comment by Fenimoure** [comment]
> I'm talking of this line 

So, do you just set it to "false"? I want this fix so bad... I don't mean that the mod's author should do that, but if that solution seems to work, I'll just do it for me and be fine
